packages/usage-reporting-protobuf/package.json

```
@@ -5,19 +5,24 @@
   "type": "module",
   "exports": {
     ".": {
-      "import": "./generated/protobuf.mjs",
-      "require": "./generated/protobuf.cjs",
-      "types": "./generated/protobuf.d.ts"
+      "types": {
+        "require": "./generated/cjs/protobuf.d.ts",
+        "default": "./generated/esm/protobuf.d.ts"
+      },
+      "import": "./generated/esm/protobuf.js",
+      "require": "./generated/cjs/protobuf.js"
     }
   },
-  "main": "generated/protobuf.cjs",
-  "module": "generated/protobuf.mjs",
-  "types": "generated/protobuf.d.ts",
+  "main": "generated/cjs/protobuf.js",
+  "module": "generated/esm/protobuf.js",
+  "types": "generated/esm/protobuf.d.ts",
   "scripts": {
-    "generate": "rm -rf generated && mkdir generated && npm run pbjs-cjs && npm run pbjs-esm && npm run pbts",
-    "pbjs-cjs": "apollo-pbjs --target static-module --out generated/protobuf.cjs --wrap commonjs --force-number --no-from-object src/reports.proto",
-    "pbjs-esm": "apollo-pbjs --target static-module --out generated/protobuf.mjs --es6 --force-number --no-from-object src/reports.proto",
-    "pbts": "mv generated/protobuf.{c,}js && apollo-pbts -o generated/protobuf.d.ts generated/protobuf.js && mv generated/protobuf.{,c}js",
+    "generate": "rm -rf generated && mkdir -p generated/{esm,cjs} && npm run pbjs-cjs && npm run pbjs-esm && npm run pbts",
+    "pbjs-cjs": "apollo-pbjs --target static-module --out generated/cjs/protobuf.cjs --wrap commonjs --force-number --no-from-object src/reports.proto",
+    "pbjs-esm": "apollo-pbjs --target static-module --out generated/esm/protobuf.mjs --es6 --force-number --no-from-object src/reports.proto",
+    "pbts-cjs": "mv generated/cjs/protobuf.{c,}js && apollo-pbts -o generated/cjs/protobuf.d.ts generated/cjs/protobuf.js && echo {\"type\":\"commonjs\"}> generated/cjs/package.json",
```

When I run this locally I end up with no double quotes in the resulting `package.json` file, making it invalid. In order to make this work correctly I had to triple escape the double quotes like so:
```suggestion
    "pbts-cjs": "mv generated/cjs/protobuf.{c,}js && apollo-pbts -o generated/cjs/protobuf.d.ts generated/cjs/protobuf.js && echo {\\\"type\\\":\\\"commonjs\\\"} > generated/cjs/package.json",
```

---

Maybe this step should just exist in the `postcompile.mjs` script alongside similar concerns (even though it doesn't fit exactly in the existing for loop). Added benefit of "compile gets run often and most importantly before publish".

```
writeFileSync(
  path.join('packages', 'usage-reporting-protobuf', 'generated', 'cjs', 'package.json'),
  JSON.stringify({ type: 'commonjs' }),
);
```

---

Alright, I went with the postcompile approach because it looks like the `echo` solution behaves differently on Windows for some reason.
I have other projects using `echo '{ \"type\": \"commonjs\" }' > dist/cjs/package.json` but it behaves differently than here. Maybe because I'm using yarn there